runtime/lib/backend/null_qubit/NullQubit.hpp

```diff
@@ -189,6 +250,10 @@ struct NullQubit final : public Catalyst::Runtime::QuantumDevice {
                         const std::vector<QubitIdType> &controlled_wires = {},
                         const std::vector<bool> &controlled_values = {})
     {
+        if (this->track_resources_) {
+            resource_data_["num_gates"]++;
+            resource_data_[name]++;
```

Comment:
Do you also need to check `Controlled` and `Adjoint` (inverse=True) ops?

Reply:
Yes this would be necessary I think. In my branch I quickly added some `C_` prefixes per control wire to account for this :)